### ISSUE TITLE
Hotkeys for new functions don't work on restart

### ISSUE THREAD

**Issue by therationalpi** [Backdash/MonikaModDev#40]
Bug report from twitter

https://twitter.com/_Wu_Kevin/status/924364543895072768

The hotkeys for 'm' 'p' and 't' are set after Monika's opening statements and/or greeting. Is there some program flow that skips them? Also, do we want to change when those are made available? Like, on subsequent restarts, should those keys be usable right away?

Alternatively, we could have some sort of feedback when hotkeys are used while Monika is busy talking? Like an 'x' pops up on the screen momentarily? In general, the feedback from those hotkeys is kind of poor.

**Comment by ThePotatoGuy** [comment]
There is a keymap skip if the user quits during the ch30_nope tag. Upon next launch, the keymaps dont get set. However:

1. On the next launch after, (if the player reaches ch30_loop), the autoload var gets set to ch30_autoload, which would set the keymaps properly.
2. I could not enter this tag normally (via deleting monika.chr). I could only get to this flow by jumping via devtools.

So I don't think this is the exact same bug that guy experienced, but there is a keymap skip.

**Comment by therationalpi** [comment]
In general, are we happy with the current behavior of these function hotkeys? Like, should we continue to wait until the player reaches ch30_loop before activating them, or should we have some other trigger point for it? Also, do we like that you can interrupt Monika mid-dialogue with this? I'm wondering if we should keep this specifically in the ch30_loop so it's only an option when Monika is otherwise quiet.

I think we should consider adding a definition for "Enable/Disable interrupts" that let's us quickly turn any of these sorts of function keys on and off, as the a program state dictates.

**Comment by ThePotatoGuy** [comment]
I agree. Setting hotkeys should be moved to definitions.rpy and an appropriate enabling/disabling function/variable should be added as well.

**Comment by therationalpi** [comment]
You know, this is making me think that maybe some new UI work could be helpful.

Like, add a button for "Write something" that does the same thing as 't' and have it get grayed out when Monika is busy. That will help conveyance on what players can do and when they can/can't use certain functions.

**Comment by haloff1** [comment]
I definitely agree with that idea.

**Comment by polishsnake** [comment]
It's not very related to the original bug, but I thought about replacing the keyword system with actual questions. So the player would go into the question asking menu (or something like that) and pick a category on where the question belongs.
Let's say the player wants to ask Monika on faith, then he/she would go into "culture" category (and in other topics there could possibly be second level of subcategories) and he/she would choose a question, for example "What's your opinion on God?" (although for this one we'd need a new topic, or possibly rewrite the one from DDLC if this example was to land in the game).

**Comment by therationalpi** [comment]
I'm telling you Polish. You've got a real talent for this:

https://github.com/Backdash/MonikaModDev/issues/126

**Comment by polishsnake** [comment]
No comment

**Comment by ThePotatoGuy** [comment]
Played around with more screens as overlays, and I came up with this:

Normal:
![buttons](https://user-images.githubusercontent.com/3499462/32760314-61d19afe-c8b3-11e7-8340-0ee5181ec398.png)

When Monika is talking:
![talking](https://user-images.githubusercontent.com/3499462/32760315-61e2fa60-c8b3-11e7-8368-034c490f5f15.png)

The buttons are custom images I created by splicing the ends of `choice_button` together, shrinking the overall length of the buttons. I also colorized a gray version of the button to signify a disabled button. 
When active, the buttons play the same sound as generic choice buttons when the mouse is hovered over them. The buttons are disabled when `allow_dialogue` is False. This can be changed easily if required.

The overlay itself is a new screen defined in `zz_hotkey_buttons.rpy`.

Code is available [here](https://github.com/ThePotatoGuy/MonikaModDev/tree/next-release-visiblehotkeybuttons)
